### PR TITLE
(DEV-3953) Adds Notes to Sermons

### PR DESCRIPTION
**Pull request by redreceipt**
## DESCRIPTION
Screen Shot 2019-08-16 at 4 44 38 PM

### What does this PR do, or why is it needed?

Combines features into a "Notes" layout with custom notes fields.

### How do I test this PR?

Go to the third sermon in the Freedom series. Take some notes. Share them.

---

> I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))

## TODO

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android if applicable
- [x] Set a relevant reviewer

## REVIEW

- [ ] Review updates to test coverage and snapshots
- [ ] Review code through the lens of being concise, simple, and well-documented

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
need to wait for an update, I need to make one small change to `ScriptureFeature` once we get them.

**Comment by redreceipt** [comment]
@vinnyjth this is what I'm doing for custom notes on a sermon...but I really don't love editing all these core files. Have any suggestions? How would you do it differently?

**Comment by vinnyjth** [comment]
What you're doing is fine - I wonder if you could clean things up a little by having a new "feature type" called `TextFeatureWithNotes` instead of injecting the custom notes into every feature.

**Comment by redreceipt** [comment]
Problem with that is displaying each feature differently. I would have to
create a "scripture feature with notes" and so on.

--
Michael Neeley
Junior Web Developer
NewSpring Church
www.newspring.cc
Cell: (803) 622-6599

**Comment by richarddubay** [comment]
@redreceipt We're just waiting on a build with this PR in it (https://github.com/ApollosProject/apollos-prototype/pull/966/files), right? Can we just pull the latest canary build of `data-connector-rock`? That should have the necessary changes in it, right?

**Comment by redreceipt** [comment]
we're waiting on this https://github.com/ApollosProject/apollos-prototype/pull/976, not 100% sure how canaries work, @vinnyjth may have insight

**Comment by redreceipt** [comment]
@vinnyjth @conrad-vanl certainly not a requirement for us but I would love either one of you to give this a quick code review. Had to extend core features here and would definitely like to know if there's a better way to go about it. Plus I'm doing some pretty complicated things here and extra eyes on the implementation would be much appreciated.

File: packages/newspringchurchapp/src/content-single/Features/SermonNotes.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  PaddedView,
+  ActionCard,
+  Touchable,
+  Icon,
+} from '@apollosproject/ui-kit';
+import share from '../../utils/content/share';
+
+const SermonNotes = ({ contentId, features }) => {
+  const [sharedMsg, changeSharedMsg] = useState('');
+  const [enhancedFeatures, enhanceFeatures] = useState([]);
```
Comment (cdanwards) [inline packages/newspringchurchapp/src/content-single/Features/SermonNotes.js:14]: Is there any way we could differentiate these names a little more?

**Review by cdanwards — COMMENTED** [review]
Do we account for commas?

Screen Shot 2019-08-20 at 2 02 44 PM

**Comment by richarddubay** [comment]
@cdanwards There is another task in the backlog (https://newspring.atlassian.net/browse/DEV-3970) with some styling tweaks that need to be made here and I think I mentioned the comma thing there.

**Comment by cdanwards** [comment]
Screen Shot 2019-08-20 at 2 17 45 PM
I did get this in the share on android. Not sure if anyone else can reproduce

**Comment by richarddubay** [comment]
Screenshot 2019-08-20 19 57 04

@cdanwards I'm not seeing that problem on Android. 🤷‍♂️

File: packages/newspringchurchapp/src/content-single/Features/SermonNotes.js (hunk 1 above)
Comment (richarddubay) [reply packages/newspringchurchapp/src/content-single/Features/SermonNotes.js:14]: What do you have in mind?

File: packages/newspringchurchapp/src/content-single/Features/SermonNotes.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        msg = `${msg + featureProps.sharing.message}\n\n`;
+        return feature;
+      }
+
+      // drop in placeholders for custom notes
+      if (featureProps.id.match(/NoteFeature/g).length > 0)
+        msg = `${msg + featureProps.id}{{}}\n\n`;
+
+      // add callbacks to swap note placeholders with custom text
+      return {
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/content-single/Features/SermonNotes.js:36]: @redreceipt @cdanwards @vinnyjth @conrad-vanl I've got to admit ... I really have no idea what this section is doing. Help?

Comment (redreceipt) [reply packages/newspringchurchapp/src/content-single/Features/SermonNotes.js:36]: 1. Adds all the non-note features
2. Adds all the placeholders: `NoteFeature123123{{}}`
3. Drops callbacks to update the placeholders inside the note feature components

Comment (richarddubay) [reply packages/newspringchurchapp/src/content-single/Features/SermonNotes.js:36]: We may have talked about this, but I'm drawing a blank. This seems so overly complex. If someone is sharing, why can't this just go through the features and get the text from each of them? Why is the addition of placeholders and whatever this `return` is doing necessary? Shouldn't this be able to just pull the text from the Note feature and include it as part of the message? What am I missing?

Comment (redreceipt) [reply packages/newspringchurchapp/src/content-single/Features/SermonNotes.js:36]: it is doing what you're describing but I'm definitely game for a simpler solution, I just couldn't think of one.

**Review by richarddubay — APPROVED** [review]
I'd like to explore a simpler solution at some point, but today is probably not that day. As best I can tell, this works, so let's go with it.

**Pull request merged**